source/common/upstream/upstream_impl.h

```diff
@@ -412,6 +412,15 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
    */
   void setOutlierDetector(const Outlier::DetectorSharedPtr& outlier_detector);
 
+  /**
+   * Wrapper around Network::Address::resolveProtoAddress() to catch the specific error in relation
+   * with envoy::api::v2::Cluster::DiscoveryType hence we can throw more meaningful message.
```

Comment:
Perhaps just `Wrapper around Network::Address::resolveProtoAddress() that provides improved error messages based on the cluster's type.`

Reply:
OK. Thank you for the suggestion. cf5a97b